napalm_panos/panos.py

```diff
@@ -0,0 +1,326 @@
+
+            self._import_file(filename)
+            # Let's load the config.
+            cmd = '<load><config><from>{0}</from></config></load>'.format(filename)
+            self.device.op(cmd=cmd)
+
+            if self.device.status == 'success':
+                self.loaded = True
+            else:
+                raise ReplaceConfigException('Error while loading config from {0}').format(filename)
+
+        else:
+            print ReplaceConfigException("This method requires a config file.")
+
+    def load_merge_candidate(self, filename=None, config=None, from_xpath=None, to_xpath=None, mode=None):
+        if filename:
+            if not from_xpath or not to_xpath or not mode:
+                raise MergeConfigException('You must provide from_xpath, '
+                                           'to_xpath and mode params for the '
+                                           'configuration to be loaded.')
+
+            elif filename.endswith('.xml') is False:
+                raise MergeConfigException('File must be in XML format.')
+
+            elif self.loaded is False:
+                if self._save_backup() is False:
+                    raise MergeConfigException('Error while storing backup '
+                                               'config.')
+            self._import_file(filename)
+
+            if self.connected is False:
+                self._open_ssh()
+
+            cmd = ("load config partial from {0} "
+                  "from-xpath {1} to-xpath {2} mode {3}".format(filename,
+                                                                from_xpath,
+                                                                to_xpath,
+                                                                mode))
+            self.connected.send_config_set([cmd])
+            self.loaded = True
+
+        elif config:
+            if self.loaded is False:
+                if self._save_backup() is False:
+                    raise MergeConfigException('Error while storing backup '
+                                               'config.')
+
+            if self.connected is False:
+                self._open_ssh()
+
+            if isinstance(config, str):
+                config = config.splitlines()
+
+            for command in config:
+                if 'set' not in command:
+                    raise MergeConfigException('Unsupported config format. '
+                                               'You should use "set" commands.'
+                                               )
+
+            self.connected.send_config_set(config)
+            self.loaded = True
+
+        else:
+            raise MergeConfigException('You must provide either an XML file '
+                                       'or a set-format string')
+
+    def compare_config(self):
+        if self.connected is False:
+            self._open_ssh()
+
+        output = self.connected.send_command("show config diff")
```

Comment:
Probably I made this tricky. Basically, I'm using PAN API most of the time to interact with the device, but I also use netmiko in some cases when needed. So, `self.device` is the reference to the API object while `self.connected` is the reference to the netmiko object. I'm going to change it and create a `self.ssh_device` to make this more clear. My bad :)